File: TenMinsPy/day88.py
# Ask for a name, output all matching csv row (exact match) searchMultiple

import csv

def search_multiple(name, file_path):
    result = []

    try:
        with open(file_path, newline='', encoding='utf-8') as csvfile:
            reader = csv.reader(csvfile)
            for row in reader:
                # Check each column for exact match
                for cell in row:
                    if cell == name:
                        result.append(','.join(row))  # Join row as string
                        break  # Stop checking this row after match

    except Exception as e:
        print(f"Error: {e}")

    return result

File: TenMinsJava/day88.java
// Ask for a name, output all matching csv row (exact match) searchMultiple
package TenMinsJava;

import java.io.File;
import java.util.ArrayList;
import java.util.Scanner;

public class day88 {

    public static String[] searchMultiple(String name) {

        ArrayList<String> result = new ArrayList<>();
        
        try {
            File file = new File("D:/coding30minsAday/textFiles/JTrialCsv.csv");
            Scanner fScan = new Scanner(file);
            
            while (fScan.hasNextLine()) {
                String line = fScan.nextLine().trim();
                String temp [] = line.split(",");
                for (String s : temp) {
                    if (s.equals(name)) {
                        result.add(line);
                    }
                }
            }

        } catch (Exception e) {
            System.err.println("Error: " + e.getMessage());
        }

        // Convert ArrayList<String> → String[]
        String[] output = new String[result.size()];
        for (int i = 0; i < result.size(); i++) {
            output[i] = result.get(i);
        }

        return output;
    }
}